pkg/postage/stamp.go

```diff
@@ -102,3 +104,25 @@ func toSignDigest(addr swarm.Address, id []byte) ([]byte, error) {
 	}
 	return h.Sum(nil), nil
 }
+
+func AttachValidStamp(batchStore Storer) func(chunk swarm.Chunk, stampBytes []byte) error {
```

Comment:
I don't understand why this is necessary. You're basically saying: "I don't trust the stamps which are persisted in localstore with the chunk". Can you please elaborate why this is needed?